Truffle updates

cds-amal opened the pull request:
This PR updates the project for use with latest version of Truffle.
  - update version of Migration.sol
  - update pragma versions 
  - insert SPDX license comment
  - rename truffle.js to truffle-config.js
  - rename network `develop` to `local` in order to allow testing with managed ganache ala `truffle test` vs having to start a separate instance of ganache.

cds-amal committed:
Update config name

cds-amal committed:
Update network

This allows for quicker testing without having to spin up ganache.

cds-amal committed:
Update Migration.sol to match Truffle's internal

cds-amal committed:
Update pragma and insert SPDX license notice

cds-amal committed:
Update tests

  - s/owner/contractOwner/g
  - add test for contract owner

cds-amal committed:
Rewrite hints

  - update link to breaking changes
  - change hints

cds-amal committed:
Update test prompt

cds-amal committed:
Enable all tests

cds-amal committed:
Update README

ConsenSys-Academy commented on README.md:
```diff
@@ -1,14 +1,23 @@
+
+ ** Check out the testing files to see how tests are implemented in JavaScript.
+    We will go over the details of implementing tests later in the course. 
+ ** If you feel there is a lot of noise, 
```
Appreciate this, you're waiting for the go-ahead to add about how to skip tests?

cds-amal replied:
oops, missed that

ConsenSys-Academy commented on contracts/SimpleBank.sol:
```diff
@@ -1,49 +1,48 @@
-    https://solidity.readthedocs.io/en/v0.6.12/060-breaking-changes.html
+    This exercise has been updated to use Solidity version 0.8.5
+    See the latest Solidity updates at
+    https://solidity.readthedocs.io/en/latest/080-breaking-changes.html
```
u r a hero

ConsenSys-Academy commented on contracts/SimpleBank.sol:
```diff
@@ -1,49 +1,48 @@
 */
-
-pragma solidity ^0.5.0;
+// SPDX-License-Identifier: MIT
```
I just checked out [the latest Solidity documentation about NatSpec comment](https://docs.soliditylang.org/en/v0.8.6/natspec-format.html) and they recommend triple slashes for comments `///`, thoughts?

cds-amal replied:
Sure, the recommendation is for NatSpec comments associated with keywords `contract`, `interface`, `function`, and `event` which can be processed to generate documentation. You'll notice the `SPDX` line in the link utilizes double slashes. 

I used the double slashes (personal choice) to indicate it is immediate instructions, or hints instead of code documentation that NatSpec should be.

Maybe there's a case for incorporating NatSpec, or maybe there isn't? I went  back and forth and and landed on having the option to ask students to create the NatSpec comments as an exercise.

ConsenSys-Academy commented on contracts/SimpleBank.sol:
```diff
@@ -1,49 +1,48 @@
     // Functions
     //
 
-    /* Use the appropriate global variable to get the sender of the transaction */
```
Curious about this, you didn't want to give the student too much initially?

cds-amal replied:
Judgement call, I think the slack/office hours support should kick in

ConsenSys-Academy commented on test/simpleBank.test.js:
```diff
@@ -8,83 +8,135 @@ installed, you can uninstall the existing version with `npm uninstall -g truffle
-  const alice = accounts[1]
-  const bob = accounts[2]
-  const deposit = web3.utils.toBN(2)
+contract("SimpleBank", function (accounts) {
```
your javascript is a thing to behold!

ConsenSys-Academy commented on test/simpleBank.test.js:
```diff
@@ -8,83 +8,135 @@ installed, you can uninstall the existing version with `npm uninstall -g truffle
-    assert.equal(expectedEventResult.withdrawAmount, withdrawAmount, "LogWithdrawal event withdrawalAmount property not emitted, check deposit method")
-  })
-})
+  it.skip("should withdraw correct amount", async () => {
```
again, this is probably vapid but your formatting is so nice

cds-amal replied:
I use prettier which autoformats the code on save.

ConsenSys-Academy commented on truffle-config.js:
```diff
@@ -1,9 +1,6 @@
-//   },
   networks: {
-    development: {
+    local: {
```
really good to know about this, so Truffle will intuit that there's no ganache running and spin up a `ganache-cli` instance?

cds-amal replied:
Yes, this is a good way to let truffle provide an in memory ganache provider. It speeds things up. I modified the travis CI script to take advantage of that

ConsenSys-Academy reviewed (commented):
@cds-amal , thank you so much for starting this, it's so needed and i'm really grateful we have you here to do it. 🙇🙇

Let me know what you think about the NatSpec comment, we tell students the final project contract should be commented according to that format.

cds-amal committed:
Update travis

  - ganache-cli is no longer needed because the project uses
    truffle's in memory provider
  - get latest version of truffle
  - checkout test from tagged version

cds-amal committed:
Skip grading tests
